app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java

```diff
@@ -0,0 +1,136 @@
+package org.fossasia.openevent.app.common.utils.ui;
+
+import android.graphics.drawable.Drawable;
+import android.support.v4.util.SparseArrayCompat;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.github.florent37.glidepalette.GlidePalette;
+
+import java.lang.ref.WeakReference;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.reactivex.functions.Consumer;
+import timber.log.Timber;
+
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public final class PaletteHolder {
+
+    private static PaletteHolder paletteHolderInstance;
```

Comment:
this is a singleton. so how will this be used for more than one palette with different images?